src/app/screens/LNURLPay/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { useState, useEffect } from "react";
+import { toast } from "react-toastify";
+import { useNavigationState } from "~/app/hooks/useNavigationState";
+import lnurlLib from "~/common/lib/lnurl";
+import getOriginData from "~/extension/content-script/originData";
+import { LNURLPayServiceResponse, OriginData } from "~/types";
+
+import LNURLPayForm from "./LNURLPayForm";
+
+const errorMessage = "Not a payRequest LNURL";
+
+const LNURLPay = () => {
```

Comment:
Is this file LNURLPay specific? 
I think this would be the same for all LNURL-* screens: we mainly load the lnurl details and then render a component.

If I read this correct this component is translating the navState to props?
why can't we directly use the navState in the form component? 
then I need to make sure that I always pass the state to `navigate()`

Reply:
LNURLPay is rendered from three different point:

- PROMPT => `origin` and `args.lnurlDetails` exist from searchParams and are passed via navigation.state [here](https://github.com/getAlby/lightning-browser-extension/pull/1257/files#diff-6aa3c99dd75ad2803ee2a158c3686f6f611f1bbaa59183f9648b97b8cf7abc10R68)
- HOME => `origin` exists but no `lnurlDetails`, therefore `origin` and `lnurl` are passed via navigation.state [here](https://github.com/getAlby/lightning-browser-extension/pull/1257/files#diff-041371f0bc0d8e5acd8d60f55e0b217ee4320404c44f5ebe9a3d61db95753a45R257-R258)
- SEND => no `origin` and no `args.lnurlDetails` exist, therefore only the `lnurl` is passed via navigation.state [here](https://github.com/getAlby/lightning-browser-extension/pull/1257/files#diff-b78594d65e4874c4016ae236fd162b02d7ea484236ce7e9b6d0c2fcd7a053f8dR42)

The component `LNURLPay` is smart enough to check what data is in the navigation.state and gets missing data accordingly to render the view.

Reply:
imo the LNURLPay component should not be rendered with different details.
in cases where no lnurlDetails is present we already loaded the lnurl details before navigating to the component - so we can just pass it there. (that's send and home)

@escapedcat then you now have another component just to not use useNavigationState() in the other one? When use useNavigationState? and when not? 

Did you think about the other LNURL* components here?